papyri/tests/test_parse.py

```diff
@@ -28,6 +34,85 @@ def test_parse_space_in_directive_section():
     )
 
 
+def test_parse_directive_body():
+    data1 = dedent(
+        """
+
+    .. directive:: Directive title
+
+        This directive declares a title and content in a block separated from
+        the definition by an empty new line.
+
+    """
+    )
+    data2 = dedent(
+        """
+
+    .. directive:: Directive title
+        This directive declares a title and content not separated by an empty
+        newline.
+
+    """
+    )
+
+    text1 = data1.strip("\n").encode()
+    text2 = data2.strip("\n").encode()
+
+    tree1 = parser.parse(text1)
+    tree2 = parser.parse(text2)
+
+    directive1 = Node(tree1.root_node).without_whitespace()
+    directive2 = Node(tree2.root_node).without_whitespace()
+
+    tsv1 = TSVisitor(text1, directive1, "test_parse_directive_body")
+    tsv2 = TSVisitor(text2, directive2, "test_parse_directive_body")
+
+    items1 = tsv1.visit(directive1)
+    items2 = tsv2.visit(directive2)
+
+    assert items1[0].name == "directive"
+    assert items1[0].args == "Directive title"
+    assert items1[0].options == dict()
+    assert (
+        items1[0].value
+        == "This directive declares a title and content in a block separated from\nthe definition by an empty new line."
+    )
+    assert items1[0].children == []
+
+    assert items2[0].name == "directive"
+    assert items2[0].args == "Directive title"
+    assert items2[0].options == dict()
+    assert (
+        items2[0].value
+        == "This directive declares a title and content not separated by an empty\nnewline."
+    )
+    assert items2[0].children == []
+
+
+def test_parse_warning_directive():
+    data = dedent(
+        """
+
+    .. warning:: Title
+
+        The warning directive does not admit a title.
+
+    """
+    )
+    text = data.strip("\n").encode()
+    tree = parser.parse(text)
+    directive = Node(tree.root_node)
+    tsv = TSVisitor(text, directive, "test_parse_directive_body")
+    new_node = directive.without_whitespace()
+    items = tsv.visit(new_node)
+
+    assert items[0].name == "warning"
+    assert items[0].args == ""
+    assert items[0].options == dict()
+    assert items[0].value == "Title The warning directive does not admit a title."
```

Comment:
Alternatively, we can raise a warning to library authors.

Reply:
Let's make a todo item in an issue once this is merged.